### ISSUE TITLE
ShortVariable rule doesn't work inside foreach

### ISSUE THREAD

**Issue by MatheusNaldi** [phpmd/phpmd#790]
- PHPMD version: PHPMD 2.8.1snapshot202002241446
- PHP Version: 7.3.18
- Installation type: phar file
- Operating System / Distribution & Version: Linux Mint 19.2 Cinnamon

### Current Behavior
It doesn't report short variables when it is inside a foreach.

### Expected Behavior
It should report short variables inside a foreach, like in the example below, it should report $n and $b variables as a short ones.

### Steps To Reproduce:
this snippet:
```php
<?php

namespace App\Test;

class Test
{
    public function testing()
    {
        $a = 1;
        echo $a;

        foreach ([1,2,3] as $n) {
            $b = $n;
            echo $b;
        }
    }
}
```

with this command:
`$ phpmd test.php text naming`

is resulting in: 
`/home/naldi/Desktop/test.php:9	Avoid variables with short names like $a. Configured minimum length is 3.`

and it's missing the lines 12 and 13 that has $n and $b short variables declared.

## Checks before submitting
* [X] Be sure that there isn't already an issue about this. See: [Issues list](https://github.com/phpmd/phpmd/issues)
* [X] Be sure that there isn't already a pull request about this. See: [Pull requests](https://github.com/phpmd/phpmd/pulls)
* [X] I have added every step to reproduce the bug.
* [X] If possible I added relevant code examples.
* [X] This issue is about 1 bug and nothing more.
* [X] The issue has a descriptive title. For example:  "JSON rendering failed on Windows for filenames with space".

**Comment by MarkVaughn** [comment]
shuffled it a bit around to make sure it's the for each part that isn't working
```
<?php

namespace App\Test;

class Test
{
    public function testing()
    {
        $array = [1, 2, 3];
        foreach ($array as $i) {
            $a = $i;
            echo $a;
        }
        $b = 'foo';

        return $b;
    }
}

```
 `phpmd test.php text naming`
.../test.php:14   Avoid variables with short names like $b. Configured minimum length is 3.

**Comment by kylekatarnls** [comment]
If anyone (from the team or not) want to help here either with a fix or just with creating a branch with a unit test to reproduce the issue, feel free to manifest. I would assign this issue to you and stay available for help.

Source code:
https://github.com/phpmd/phpmd/blob/master/src/main/php/PHPMD/Rule/Naming/ShortVariable.php

Tests:
https://github.com/phpmd/phpmd/blob/master/src/test/php/PHPMD/Rule/Naming/ShortVariableTest.php
`->getClass()` in test files refer to the class with the same name as the test method in the matching file from:
https://github.com/phpmd/phpmd/tree/master/src/test/resources/files/Rule/Naming/ShortVariable

**Comment by artem-prozorov** [comment]
@kylekatarnls Hi! I'm up for this, will do within a week or so
